packages/plugin-legacy/src/index.ts

```diff
@@ -671,6 +669,18 @@ async function buildPolyfillChunk(
           manualChunks: undefined
         }
       }
+    },
+    // Don't run esbuild for transpilation or minification
+    // because we don't want to transpile code.
+    esbuild: false,
+    optimizeDeps: {
+      esbuildOptions: {
+        // If a value above 'es5' is set, esbuild injects helper functions which uses es2015 features.
+        // This limits the input code not to include es2015+ codes.
+        // But core-js is the only dependency which includes commonjs code
+        // and core-js doesn't include es2015+ codes.
+        target: 'es5'
+      }
     }
```

Comment:
Following up of the convo in #9635, I think `esbuild: false` make sense since you showed that core-js is already in es5, and same for systemjs (except it's now using `const`).

I don't quite understand the `optimizeDeps.esbuildOptions.target` part though since `optimizeDeps` is only for dev only for now, unless you're anticipating when we optimize in build too? I'd actually slightly lean to not optimize in build too then to simplify the bundling process 🤔

Reply:
Ah, yep, we could remove this for now.
#8660 was introduced when optimize in build was enabled by default. We'll need to add this or enable commonjs plugin when optimize in build is enabled by default.

Reply:
Actually thinking now, maybe it wouldn't hurt leaving it in anyways, but it's definitely something to keep in mind when we completely remove the commonjs plugin